### ISSUE TITLE
no-ignored-return coordination eslint，No error message

### ISSUE THREAD

**Issue by jymlbei** [SonarSource/eslint-plugin-sonarjs#353]
![image](https://user-images.githubusercontent.com/25196960/169660631-c514feb0-a29b-4553-b9d7-6d0f38f81932.png)
There are no error messages here，when Eslint is on
But in sonar
![image](https://user-images.githubusercontent.com/25196960/169660689-ee213f57-f241-4cd8-9880-5c41204ca73e.png)
It can be verified

How can I make the editor display errors

.eslintrc.js
![image](https://user-images.githubusercontent.com/25196960/169660722-807a6e6f-d85b-47de-ab2f-751974d6c1b4.png)

**Comment by vilchik-elena** [comment]
How about installing SonarLint?
Apart from that I hardly can help you. Try usual debugging: is any other rule from ESLint is reporting in IDE? does the issue appear if you are running eslint in command line? etc. Narrow the problem

**Comment by ben12** [comment]
Same for me.

It works with eslint if I downgrade @typescript-eslint/parser from latest v6 to latest v5.
eslint-plugin-sonarjs require an upgrade of @typescript-eslint/parser.

**Comment by Chris380** [comment]
I also do not get warnings and errors for `no-ignored-return` in VS Code and command line, but I do get them (in both) for other rules like `sonarjs/prefer-immediate-return`. Maybe the configuration in my little demo is incorrect.

package.json
```json
{
  "name": "sonar-test",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "author": "",
  "license": "ISC",
  "dependencies": {
    "eslint-plugin-sonarjs": "^0.21.0"
  },
  "devDependencies": {
    "@typescript-eslint/eslint-plugin": "^5",
    "@typescript-eslint/parser": "^5",
    "eslint": "^8.51.0",
    "typescript": "5.1"
  }
}
```

.eslintrc.json
```json
{
    "env": {
        "browser": true,
        "es6": true,
        "node": true
    },
    "extends": [
        "eslint:recommended",
        "plugin:@typescript-eslint/recommended",
        "plugin:sonarjs/recommended"
    ],
    "parser": "@typescript-eslint/parser",
    "parserOptions": {
      "ecmaVersion": 2018,
      "ecmaFeatures": { "modules": true },
      "sourceType": "module"
    },
    "plugins": [
        "@typescript-eslint",
        "sonarjs"
    ],
    "rules": {
        "sonarjs/prefer-immediate-return": "warn",
        "sonarjs/no-ignored-return": "error",
        "sonarjs/no-duplicate-string": ["warn", { "threshold": 2 }]
    }
}
```

In my demo.ts the "immediately return" produces a warning, the lastIndexOf does not:
![image](https://github.com/SonarSource/eslint-plugin-sonarjs/assets/14816381/e120c8e9-f2ef-48bb-9683-6d4c40c4e1f2)

![image](https://github.com/SonarSource/eslint-plugin-sonarjs/assets/14816381/eed7ee9e-80c0-4a9f-9531-66da0bedf6a7)

Thanks for your help

**Comment by Wohops** [comment]
This issue has been migrated to Jira. [ESLINTJS-22](https://sonarsource.atlassian.net/browse/ESLINTJS-22)
